Modules/Findgfsio.cmake

```diff
@@ -12,8 +12,8 @@ if(DEFINED ENV{GFSIO_LIB4} )
 
   set(kinds "4")
   foreach(kind ${kinds})
-    set(lib_name ${name}_${kind})
-    set(versioned_lib_name ${name}_${version}_${kind})
+    set(lib_name ${name})
+    set(versioned_lib_name ${name}_${version}_${kinds})
```

Comment:
This is different then this git diff:
```
diff --git a/Modules/Findgfsio.cmake b/Modules/Findgfsio.cmake
index 44b221a..7b74e49 100644
--- a/Modules/Findgfsio.cmake
+++ b/Modules/Findgfsio.cmake
@@ -1,7 +1,7 @@
 # This module looks for environment variables detailing where GFSIO lib is
 # If variables are not set, GFSIO will be built from external source
 if(DEFINED ENV{GFSIO_LIB4} )
-  set(GFSIO_LIB4 $ENV{GFSIO_LIB4} CACHE STRING "GFSIO Library Location" )
+  set(GFSIO_LIB4 $ENV{GFSIO_LIB4} CACHE STRING "GFSIO_4 Library Location" )
   set(GFSIO_INC4 $ENV{GFSIO_INC4} CACHE STRING "GFSIO_4 Include Location" )
 
   set(name "gfsio")
@@ -17,11 +17,11 @@ if(DEFINED ENV{GFSIO_LIB4} )
 
     if(EXISTS ${${uppercase_name}_LIB${kind}} )
       get_filename_component(lib_dir ${${uppercase_name}_LIB${kind}} DIRECTORY)
-      find_library(gfsio_path NAMES ${versioned_lib_name} PATHS ${lib_dir} NO_DEFAULT_PATH)
+      find_library(gfsio_path_${kind} NAMES ${versioned_lib_name} PATHS ${lib_dir} NO_DEFAULT_PATH)
    
       add_library(${lib_name} STATIC IMPORTED)
       set_target_properties(${lib_name} PROPERTIES
-        IMPORTED_LOCATION ${gfsio_path}
+        IMPORTED_LOCATION ${gfsio_path_${kind}}
         INTERFACE_INCLUDE_DIRECTORIES ${${uppercase_name}_INC${kind}})
     endif()
   endforeach()
```

Reply:
I had pulled the ${kind} extension off altogether for the gfsio_path since there is only one kind for gfsio, but I will put it back.

Reply:
and I think that is the right way to go. Since we are only building _4, remove all references to _4 and replace _LIB4 with LIB and _INC4 with INC, similar to the nemsio library.

This half baked hack is bound to create confusion and cause issues down the road. 

Let me know if you do this, and I can make the change in the gfsio library to export the correct interfaces.